Add EIP: Reduced Attestation Format (#8372)
ethereum/EIPs: EIPS/eip-7610.md

---

> eip: 7610

Reviewer: ```suggestion
eip: 7670
```
Assigning next sequential EIP/ERC/RIP number.
Numbering changed to sequential from 7500.

Please also update the filename.

title: Reduced Attestation Format
description: <Description is one full (short) sentence>
author: Giulio Rebuffo (@Giulio2002)

> discussions-to: https://ethereum-magicians.org/t/reduced-attestation-format-for-gossiping-attestations/19458

Reviewer: ```suggestion
discussions-to: https://ethereum-magicians.org/t/eip-7670-reduced-attestation-format-for-gossiping-attestations/19458
```

status: Draft
type: Standards Track
category: Core
created: 2024-03-31
---

## Abstract

> This EIP introduces a reduced version of the `Attestation`sformat to reduce the size of attestations transmitted on the Consensus Layer's gossip network. This is achieved by replacing Checkpoint objects in gossiped `AttestationData`s with their respective 8-byte digests.

Reviewer 2: ```suggestion
This EIP introduces a reduced version of the `Attestation` format to reduce the size of attestations transmitted on the Consensus Layer's gossip network. This is achieved by replacing Checkpoint objects in gossiped `AttestationData` with their respective 8-byte digests.
```
imo looks cleaner this way

## Motivation

> Attestations currently dominate the bandwidth usage of the Consensus Layer's gossip network (approx. 90% of the traffic). By optimizing the size of `AttestationData`, the aim is to reduce network load to improve node performance, and, most importantly, allow for greater throughput of other significant consensus data, such as Blobs. As a matter of fact, the current `Attestation` structure is not optimized for network transmission, leading to unnecessarily high bandwidth consumption. As a matter of fact, the application of Snappy block compression only reduces the size of an attestation by less than 10%.

Reviewer 2: ```suggestion
Attestations currently dominate the bandwidth usage of the Consensus Layer's gossip network (approx. 90% of the traffic). By optimizing the size of `AttestationData`, the aim is to reduce network load to improve node performance, and, most importantly, allow for greater throughput of other significant consensus data, such as Blobs. As a matter of fact, the current `Attestation` structure is not optimized for network transmission, leading to high bandwidth consumption. As a matter of fact, the application of Snappy block compression only reduces the size of an attestation by less than 10%.
```
+ may be a diagram with subnets subscribed with traffic a node experiences will provide a very nice context

Author: I do not actually have that for now, will work on it this week.

Reviewer 2: ok let me know if you want to add that or you think this EIP is ready for draft?

## Specification

### Constants

```
CHECKPOINT_DIGEST_LENGTH = 8
```

### Containers

```python
class ReducedSignedAggregateAndProof(Container):
    message: ReducedAggregateAndProof
    signature: BLSSignature
```

```python
class ReducedAggregateAndProof(Container):
    aggregator_index: ValidatorIndex
    aggregate: ReducedAttestation
    selection_proof: BLSSignature
```

```python
class ReducedAttestation(Container):
    aggregation_bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE]
    data: ReducedAttestationData
    signature: BLSSignature
```

```python
class ReducedAttestationData(Container):
    class AttestationData(Container):
    slot: Slot

>     index: CommitteeIndex

Reviewer 2: index is being pulled out of attestation data: https://eips.ethereum.org/EIPS/eip-7549

>     index: CommitteeIndex

Reviewer 2: index is being pulled out of attestation data: https://eips.ethereum.org/EIPS/eip-7549
